src/main/java/github/scarsz/discordsrv/DiscordSRV.java

```diff
@@ -1728,6 +1728,77 @@ public void sendLeaveMessage(Player player, String quitMessage) {
         }
     }
 
+    
+    /**
+     * Gives an "online" role to a player after joining. The role id can be added in the config.
+     *
+     * @param player the player
+     * @see #removePlayerOnlineRole(Player)
+     */
+    public void givePlayerOnlineRole(Player player) {
+        if (player == null) throw new IllegalArgumentException("player cannot be null");
+
+        // retrieves the role if possible, if not returns null
+        String roleName = DiscordSRV.config().dget("OnlineRoleName").asString();
+        Role role = DiscordUtil.getRoleByName(getMainGuild(), roleName);
+        if (role == null){
+            getLogger().severe("Role \""+roleName+"\" could not be found. Make sure OnlineRoleName is correctly set in the config.yml");
+            return;
+        }
+
+        // retrieves the discord id, if possible. If not, returns null
+        String discordId = accountLinkManager.getDiscordId(player.getUniqueId());
+        if (discordId == null) {
+            getLogger().severe("Online role is enabled, but discord linking is not.");
+            return;
+        }
+
+        try {
+            getMainGuild().addRoleToMember(discordId, role).queue();
+        } catch (PermissionException e) {
+            if (e.getPermission() != Permission.UNKNOWN) {
+                DiscordSRV.warning("Could not add " + player.getDisplayName() + " to role " + role + " because the bot does not have the \"" + e.getPermission().getName() + "\" permission");
+            } else {
+                DiscordSRV.warning("Could not add " + player.getDisplayName() + " to role " + role + " because \"" + e.getMessage() + "\"");
+            }
+        }
+    }
+
+    /**
+     * Removes an "online" role to a player after leaving. The role id can be added in the config.
+     *
+     * @param player the player
+     * @see #givePlayerOnlineRole(Player)
+     */
+    public void removePlayerOnlineRole(Player player)  {
+        if (player == null) throw new IllegalArgumentException("player cannot be null");
+
+        // retrieves the role if possible, if not returns null
+        String roleName = DiscordSRV.config().dget("OnlineRoleName").asString();
+        Role role = DiscordUtil.getRoleByName(getMainGuild(), roleName);
+        if (role == null){
+            getLogger().severe("Role \""+roleName+"\" could not be found. Make sure OnlineRoleName is correctly set in the config.yml");
+            return;
+        }
+
+        // retrieves the discord id, if possible. If not, returns null
+        String discordId = accountLinkManager.getDiscordId(player.getUniqueId());
+        if (discordId == null) {
+            getLogger().severe("Online role is enabled, but discord linking is not.");
```

Comment:
Wrong logger method, invalid message (discord linking is not disableable, message *should* say that they don't have a linked Discord account)